src/coreclr/nativeaot/System.Private.CoreLib/src/System/Threading/SyncTable.cs

```diff
@@ -77,11 +78,21 @@ internal static class SyncTable
         private static int s_freeEntryList;
 
         /// <summary>
-        /// The index of the lowest never used entry.  We skip the 0th entry and start with 1.
+        /// The index of the lowest never used entry.  We skip the 0th and 1st entries and start with 2.
         /// If all entries have been used, s_unusedEntryIndex == s_entries.Length.  This counter
         /// never decreases.
         /// </summary>
-        private static int s_unusedEntryIndex = 1;
+        private static int s_unusedEntryIndex = 2;
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static ref Entry EntryRef(int i)
```

Comment:
Are the bounds-check that expensive everywhere? I would rename it to `UnsafeEntryRef` and only use it in the two perf-critical places.

Reply:
Bounds check adds some code into the caller and reads memory at arbitrary distance from the location that we need. 
It makes difference only in two critical places though. We can use indexer in others.